pkg/envtest/server.go

```diff
@@ -70,38 +71,54 @@ type Environment struct {
 
 	// CRDDirectoryPaths is a list of paths containing CRD yaml or json configs.
 	CRDDirectoryPaths []string
+
+	// UseExisting indicates that this environments should use an
+	// existing kubeconfig, instead of trying to stand up a new control plane.
+	// This is useful in cases that need aggregated API servers and the like.
+	UseExistingCluster bool
 }
 
 // Stop stops a running server
 func (te *Environment) Stop() error {
+	if te.UseExistingCluster {
+		return nil
+	}
 	return te.ControlPlane.Stop()
 }
 
 // Start starts a local Kubernetes server and updates te.ApiserverPort with the port it is listening on
 func (te *Environment) Start() (*rest.Config, error) {
-	te.ControlPlane = integration.ControlPlane{}
-	te.ControlPlane.APIServer = &integration.APIServer{Args: defaultKubeAPIServerFlags}
-	if os.Getenv(envKubeAPIServerBin) == "" {
-		te.ControlPlane.APIServer.Path = defaultAssetPath("kube-apiserver")
-	}
-	if os.Getenv(envEtcdBin) == "" {
-		te.ControlPlane.Etcd = &integration.Etcd{Path: defaultAssetPath("etcd")}
-	}
-	if os.Getenv(envKubectlBin) == "" {
-		// we can't just set the path manually (it's behind a function), so set the environment variable instead
-		if err := os.Setenv(envKubectlBin, defaultAssetPath("kubectl")); err != nil {
+	if te.UseExistingCluster && te.Config == nil {
```

Comment:
hmm.. I tried moving the `te.Config == nil` inside the `if te.UseExistingCluster` block. It is simple to read: WDYT ?

```Go
  if te.UseExistingCluster {
       if te.Config == nil {
         te.Config, err = config.GetConfig()
         // 
      } else {
       // use the supplied config 
     }
   } else {
      // ....
   }
```

Reply:
I tested the code by cutting and pasting it into my vendor directory.  
It worked as it should.

I was unable to modify the Gopkg.toml per instructions, as it encountered other dependencies.
```
dep ensure
Solving failure: No versions of sigs.k8s.io/controller-runtime met constraints:
	master: Could not introduce sigs.k8s.io/controller-runtime@master, as it is not allowed by constraint 5baed9d8a00f6ef3896f733af266fafbfee92d35 from project github.com/jkevlin/demo.
	v0.1.1: Could not introduce sigs.k8s.io/controller-runtime@v0.1.1, as it is not allowed by constraint 5baed9d8a00f6ef3896f733af266fafbfee92d35 from project github.com/jkevlin/demo.
	master: Could not introduce sigs.k8s.io/controller-runtime@master, as it is not allowed by constraint 5baed9d8a00f6ef3896f733af266fafbfee92d35 from project github.com/jkevlin/demo.
	admissionwebhook: Could not introduce sigs.k8s.io/controller-runtime@admissionwebhook, as it is not allowed by constraint 5baed9d8a00f6ef3896f733af266fafbfee92d35 from project github.com/jkevlin/demo.
	bulk_deposit: Could not introduce sigs.k8s.io/controller-runtime@bulk_deposit, as it is not allowed by constraint 5baed9d8a00f6ef3896f733af266fafbfee92d35 from project github.com/jkevlin/demo.
	review: Could not introduce sigs.k8s.io/controller-runtime@review, as it is not allowed by constraint 5baed9d8a00f6ef3896f733af266fafbfee92d35 from project github.com/jkevlin/demo.
```

Reply:
I believe that @droot 's suggested changes are correct, because supplying a te.Config should not eliminate the ability to UseExistingCluster?

Reply:
The stanza in `Gopkg.toml` should look something like

```toml
[[constraint]]
name="sigs.k8s.io/controller-runtime"
source="github.com/directxman12/controller-runtime"
branch="feature/external-cluster-envtest"
```

Reply:
yeah, that variant is fine too.  I'll make the change.  Went back-and-forth anyway.